src/main/java/net/dv8tion/jda/internal/entities/MemberImpl.java

```diff
@@ -53,7 +55,9 @@ public MemberImpl(GuildImpl guild, User user)
         this.user = user;
         JDAImpl jda = (JDAImpl) getJDA();
         boolean cacheState = jda.isCacheFlagSet(CacheFlag.VOICE_STATE) || user.equals(jda.getSelfUser());
+        boolean cacheOnline = jda.isCacheFlagSet(CacheFlag.CLIENT_STATUS);
         this.voiceState = cacheState ? new GuildVoiceStateImpl(this) : null;
+        this.clientStatus = cacheOnline ? new ConcurrentHashMap<>(5) : null;
```

Comment:
Where does this `5` come from?

Reply:
Shouldn't this be something more like `ClientType.values().length` ?

Reply:
The default capacity is usually 16 so this is intended lower the memory footprint since we know its not more then 3 types but there might come a few more in the future.